hamocc/mo_bgcmean.F90

```diff
@@ -5,2354 +5,2217 @@
+    !
+    ! Local varaibles
+    integer :: i,j,l,o
+    !
+    ! --- Check whether field should be accumulated
+    do o=1,nbgc
+      if (pos(o).EQ.0) cycle
+      !
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            bgct_bur(i,j,pos(o))=bgct_bur(i,j,pos(o))+fld(i,j)
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+    enddo
+    !
+  end subroutine accbur
+
+
+  subroutine finsrf(posacc,poswgt)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: finalise accumulation of weighted 2d fields
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
+    integer, intent(in)  :: posacc ! position of accumulated field in buffer
+    integer, intent(in)  :: poswgt ! position of accumulated weights
+    !
+    ! Local variables
+    integer :: i,j,l
+    real, parameter :: epsil=1e-11
+    !
+    ! --- Check whether field should be initialised
+    if (posacc.EQ.0) RETURN
+    !
+    !$OMP PARALLEL DO PRIVATE(l,i)
+    do j=1,jj
+      do l=1,isp(j)
+        do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+          bgcm2d(i,j,posacc)=bgcm2d(i,j,posacc)/max(epsil,bgcm2d(i,j,poswgt))
+        enddo
+      enddo
+    enddo
+    !$OMP END PARALLEL DO
+    !
+  end subroutine finsrf
+
+
+
+  subroutine finlyr(posacc,poswgt)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: finalise accumulation of weighted 3d layer fields
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
+    integer, intent(in)  :: posacc ! position of accumulated field in buffer
+    integer, intent(in)  :: poswgt ! position of accumulated weights
+    !
+    ! Local variables
+    integer :: i,j,k,l
+    real, parameter :: epsil=1e-11
+    !
+    ! --- Check whether field should be initialised
+    if (posacc.EQ.0) RETURN
+    !
+    do k=1,kk
+      !$OMP PARALLEL DO PRIVATE(l,i)
+      do j=1,jj
+        do l=1,isp(j)
+          do i=max(1,ifp(j,l)),min(ii,ilp(j,l))
+            if (bgcm3d(i,j,k,poswgt).GT.epsil) then
+              bgcm3d(i,j,k,posacc)=bgcm3d(i,j,k,posacc)/bgcm3d(i,j,k,poswgt)
+            else
+              bgcm3d(i,j,k,posacc)=nf90_fill_double
+            endif
+          enddo
+        enddo
+      enddo
+      !$OMP END PARALLEL DO
+    enddo
+    !
+  end subroutine finlyr
+
+
+
+  subroutine wrtsrf(pos,frmt,sfac,offs,cmpflg,vnm)
+    !
+    ! --- ------------------------------------------------------------------
+    ! --- Description: writes diagnostic 2d field to file
+    ! --- ------------------------------------------------------------------
+    !
+    ! Arguments
+    integer, intent(in) :: pos          ! variable position in common buffer
+    real,    intent(in) :: frmt         ! format/precision of output
```

Comment:
Dear @mvertens , here and below: it seems as if the the type of `frmt` and `offs` got mixed up. (requires: `real :: offs`, `integer :: frmt` )

Reply:
Sorry - I've been recompiling frequently - but missed doing it for this. Should be fixed now.

Reply:
Could you push the changes?